cmake/nsis/CMakeLists.txt

```
@@ -4,9 +4,9 @@ if(LMMS_MSVC_YEAR)
 endif()
 
 SET(CPACK_PACKAGE_ICON                  "${CMAKE_SOURCE_DIR}/cmake/nsis/nsis_branding.bmp")
-IF(MSVC)
+IF(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 	STRING(REPLACE "/" "\\\\" CPACK_PACKAGE_ICON ${CPACK_PACKAGE_ICON})
-ENDIF(MSVC)
+ENDIF()
 SET(CPACK_NSIS_MUI_ICON                 "${CMAKE_SOURCE_DIR}/cmake/nsis/icon.ico")
```

AFAIR, `CMAKE_SYSTEM_NAME` is always `"Windows"` when we're building with NSIS.

```suggestion
STRING(REPLACE "/" "\\\\" CPACK_PACKAGE_ICON ${CPACK_PACKAGE_ICON})
```

---

> Why are you removing the set(icon) line from below?

Good catch.  GitHub doesn't like commit suggestions with removed lines so I had to do it manually.  Fixed.